src/lib/tradeExecution.ts

```diff
@@ -120,6 +121,27 @@ export class TradeExecution {
 
       const { cancelPolling } = poll({
         fn: async () => {
+          if (getConfig().VITE_FEATURE_ACTION_CENTER) {
+            const swap = selectCurrentSwap(store.getState())
+
+            if (!swap) return
+
+            const payload: StatusArgs = {
+              stepIndex,
+              status: txStatusBySwapStatus[swap.status],
+              message: swap.statusMessage,
+              buyTxHash: swap.buyTxHash,
+            }
+            this.emitter.emit(TradeExecutionEvent.Status, payload)
+
+            if (swap.status === SwapStatus.Success)
+              this.emitter.emit(TradeExecutionEvent.Success, payload)
+            if (swap.status === SwapStatus.Failed)
+              this.emitter.emit(TradeExecutionEvent.Fail, payload)
+
+            return txStatusBySwapStatus[swap.status]
+          }
+
```

Comment:
We are probably going to keep this is we keep the action center flag in

But we could remove the poller and/or checkTradeStatus call under these lines and keep this as it is later!

Now the action center is source of truth if we have the action center activated, no more request dup!